internal/k8s/statusaddress.go

```diff
@@ -168,22 +168,6 @@ func (s *StatusAddressUpdater) OnAdd(obj interface{}) {
 			}
 		}
 
-		// Only set the Gateway's address if it has a condition of "Ready: true".
-		var ready bool
-		for _, cond := range o.Status.Conditions {
-			if cond.Type == string(gatewayapi_v1alpha2.GatewayConditionReady) && cond.Status == metav1.ConditionTrue {
-				ready = true
-				break
-			}
-		}
-		if !ready {
-			s.Logger.
-				WithField("name", o.Name).
-				WithField("namespace", o.Namespace).
-				Debug("Gateway is not ready, not setting address")
-			return
-		}
-
```

Comment:
I can't remember why I included this logic here, but it creates an issue in the case where an address is requested in the Gateway spec, because it's considered "Not Ready" until that address shows up in the status, which this would block. Tests seem OK without it so hoping its removal doesn't cause issues..